parsl/monitoring/db_manager.py

```diff
@@ -414,6 +484,80 @@ def start(self,
                                 if task_try_id in deferred_resource_messages:
                                     reprocessable_first_resource_messages.append(
                                         deferred_resource_messages.pop(task_try_id))
+                        elif msg_type == MessageType.FILE_INFO:
+                            file_id = msg['file_id']
+                            file_all_messages.append(msg)
+                            if file_id in inserted_files:
+                                changed = False
+                                # once certain items are set, they should not be changed
+                                if inserted_files[file_id]['timestamp'] is None:
```

Comment:
there's a lot of logic here dealing with "when presented with two versions of the truth, I'm going to arbitrarily pick one of them and discard the other".  All these if-statements could go away if you arbitrarily picked the most recent version of the truth, rather than the oldest-except-None version of the truth, I think, which would be simpler code. In both cases, there is a consistency-of-truth violation that's being silently ignored.